project-api/project-api-test/src/test/java/io/ballerina/projects/test/PackageResolutionLocalTests.java

```diff
@@ -0,0 +1,431 @@
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+    }
+
+    @Test(description = "Tests local packages (published, but patched with the same version) as direct dependencies",
+            dependsOnMethods = "testCase0001")
+    public void testCase0002(ITestContext ctx) throws IOException {
+
+        /*
+            Pushed package_a to central
+            Changed write method module, built again and pushed to local repository with 'same' version
+            Built project_0002
+                package_a resolved from local repository since it mentioned in the Ballerina.toml
+                package_a resolved from central and built successfully and generated jars when dependency removed
+                from Ballerina.toml
+         */
+
+        Path projectDirPath = PROJECTS_DIRECTORY.resolve("project_0002");
+        ctx.getCurrentXmlTest().addParameter("packagePath", String.valueOf(projectDirPath));
+
+        // project_002 --> package_a:0.0.1
+        // Push package_a:0.0.1 to central
+        cacheDependencyToCentralRepository(PACKAGES_DIRECTORY.resolve("package_a_0_0_1_central"));
+        // Check if its actually cached to central properly
+        Path centralCache = USER_HOME.resolve(ProjectConstants.REPOSITORIES_DIR)
+                .resolve(CENTRAL_REPOSITORY_CACHE_NAME).resolve(ProjectConstants.BALA_DIR_NAME).resolve(ORG).resolve(
+                        "package_a").resolve("0.0.1").resolve("java11").resolve("dependency-graph.json");
+        Assert.assertTrue(centralCache.toFile().exists());
+
+        // Push package_a:0.0.1 to local
+        // package_a is already in local
+        // Check if its actually cached to local properly
+        Path localCache = USER_HOME.resolve(ProjectConstants.REPOSITORIES_DIR)
+                .resolve(LOCAL_REPOSITORY_NAME).resolve(ProjectConstants.BALA_DIR_NAME).resolve(ORG).resolve(
+                        "package_a").resolve("0.0.1").resolve("java11").resolve("dependency-graph.json");
+        Assert.assertTrue(localCache.toFile().exists());
+
+        // Build project_0002:1.0.0
+        BuildProject buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        failIfDiagnosticsExists(buildProject);
+
+        Assert.assertEquals(readFileAsString(projectDirPath.resolve(DEPENDENCIES_TOML)),
+                readFileAsString(projectDirPath.resolve(RESOURCE_DIR_NAME).resolve("Dependencies.toml")));
+
+        // If the user removes the local dependency from the Ballerina.toml
+        projectDirPath = PROJECTS_DIRECTORY.resolve("project_0002_2");
+
+        // Build project_0002:1.0.0
+        buildProject = BuildProject.load(projectEnvironmentBuilder, projectDirPath);
+        buildProject.save();
+        // expectedDiagnostics(buildProject, 1, "undefined function 'func1'");
+        failIfDiagnosticsExists(buildProject);
```

Comment:
How do we make sure that the package is picked from local repo vs central cache?  I think we should add a breaking change to `package_a` when caching to the local and then assert the diagnostic here